qiskit/pulse/builder.py

```diff
@@ -1730,6 +1730,63 @@ def barrier(*channels_or_qubits: Union[chans.Channel, int]):
 
 
 # Macros
+def macro(func: Callable):
+    """Wrap a Python function and activate the parent builder context at calling time.
+
+    This enables embedding Python functions as builder macros.
```

Comment:
can you add description about backend insertion from parent context?

Reply:
Addressed in c7eced8.